src/OrchardCore/OrchardCore.Users.Abstractions/Services/IUserToExternalLoginProvider.cs

```diff
@@ -0,0 +1,49 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+
+namespace OrchardCore.Users;
+
+/// <summary>
+/// Interface <c>IUserToExternalLoginProvider</c> allow to create services used to decide
+/// when link an existing local user account to the login informations that comes from an external
+/// login system like OpenId or GitHub.
+/// </summary>
+public interface IUserToExternalLoginProvider
+{
+    /// <summary>
+    /// Method <c>CanHandle</c> establish if the service that implement this interface can handle
+    /// this kind of external login.
+    /// </summary>
+    /// <param name="info">
+    /// external login information with type and other data to establish if
+    /// service can handle this kind of external login.
+    /// </param>
+    /// <returns>
+    /// True, if the service can handle the external login by the informations in the parameter. False, instead.
+    /// </returns>
+    bool CanHandle(ExternalLoginInfo info);
+
+    /// <summary>
+    /// Method <c>GetUserAsync</c> return a local user account that match the login data in the parameter.
+    /// </summary>
+    /// <param name="info">
+    /// external login information, used to establish if exists a local user account that match login data.
+    /// </param>
+    /// <returns>
+    /// an object that implement <c>IUser</c>, if the exist a local account that match the external login data. Null, instead.
+    /// </returns>
```

Comment:
```suggestion
    /// <summary>
    /// Retrieves a local user account that corresponds to the provided external login information, if one exists.
    /// </summary>
    /// <param name="info">The external login information.</param>
    /// <returns>
    /// An instance of <see cref="IUser" /> if there's a local account matching the external login data; otherwise, null.
    /// </returns>
```